lib/utils/device/api.ts

```diff
@@ -256,24 +261,35 @@ export class DeviceAPI {
 			}
 		}
 
-		return new Promise((resolve, reject) => {
-			return request(opts, (err, response, body) => {
-				if (err) {
-					return reject(err);
-				}
-				switch (response.statusCode) {
-					case 200:
-						return resolve(body);
-					case 400:
-						return reject(new ApiErrors.BadRequestDeviceAPIError(body.message));
-					case 503:
-						return reject(
-							new ApiErrors.ServiceUnavailableAPIError(body.message),
-						);
-					default:
-						return reject(new ApiErrors.DeviceAPIError(body.message));
-				}
+		const doRequest = async () => {
```

Comment:
Just wrapping existing code in a `doRequest` function that is then retried a number of times.